packages/next/src/server/next-server.ts

```diff
@@ -333,8 +337,8 @@ export default class NextNodeServer extends BaseServer<
   protected async prepareImpl() {
     await super.prepareImpl()
 
-    if (this.instrumentation) {
-      await this.instrumentation.register?.()
+    if (this.isProductionInstrumentationEnabled()) {
+      await this.instrumentation?.register?.()
```

Comment:
That's a bit confusing now. Why don't we keep the previous condition, and remove `this.runInstrumentationHookIfAvailable()` from `NextDevServer#prepareImpl` instead? Or couldn't we do that because of the `'run-instrumentation-hook'` tracing in the dev server?

Reply:
We could do something like this, I guess?

```ts
await this.startServerSpan
  .traceChild('run-instrumentation-hook')
  .traceAsyncFn(() => super.prepareImpl())
```

Maybe not the best trace name then.